pkg/locker/locker.go

```diff
@@ -90,6 +93,13 @@ func (l *Locker) Lock(name string) {
 	// once locked then we can decrement the number of waiters for this lock
 	nameLock.Lock()
 	nameLock.dec()
+
+	if nameLock.err != nil {
```

Comment:
Why not check for nameLock.err before the waiter count is incremented at line 85?